### PR TITLE
NF: Add more details to Kotlin Migration

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
Some details were not clear during my first try of this feature.

In particular, I didn't have the "commit" window. Instead I had a different
non-modal window, which did not offer as many option. So I had to explain how to
get the modal window.

I also added instructions about what to do if build fail, at the place where we
expect failure.

I also added sub-sections. This way, if the process fail, it's easier to know
where we are.

### CODE REVIEW

**Review by mikehardy — APPROVED** [review]
LGTM, for what it's worth, but David has been mowing through these more than I

File: docs/kotlin-migration.md [hunk 1]
```diff
@@ -8,16 +8,85 @@ We have developed a nonstandard process to convert existing files to Kotlin. Rat
...
+* Right click on the bottom commit "rename .java to .kt"
+* Click on "Interactively Rebase from here"
+* Right click on the top commit "Rename .java to .kt"
+* Click on "Stope To Edit"
+* Click on "Start rebasing"
+
+#### Edit the first commit
+
+* Open the file `kotlinMigration.gradle`
+* Find the line `def className = ""`.
```
Comment (david-allison-1) [inline docs/kotlin-migration.md:55]: This should mention setting `def source = Source.MAIN` to `Source.TEST` / `Source.ANDROID_TEST` if the file was not in `/main`

File: docs/kotlin-migration.md [hunk 2]
```diff
@@ -8,16 +8,85 @@ We have developed a nonstandard process to convert existing files to Kotlin. Rat
...
+
+### Edit the second commit
+
+#### Reverting kotlinMigration.gradle
+* Open `kotlinMigration.gradle` if it is not already opened.
+* Find the line you edited. The one that starts with `def className = "`.
+* Replace this line by `def className = ""` (that is: replace the path of the file and set the variable to an empty string)
+
+#### Amending the second commit
+If you already know how to amend in git, great. Otherwise, follow those steps:
```
Comment (david-allison-1) [inline docs/kotlin-migration.md:82]: This shouldn't be copy/pasted and should be moved to an appendix

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
Looks good! Great to have a second pair of eyes on this

A few rewordings (all optional), I'd like to move the added standard git operations to an appendix to remove duplication and so an experienced user won't be bogged down by irrelevant content.

File: docs/kotlin-migration.md (hunk 1 above)
Comment (Arthur-Milchior) [reply docs/kotlin-migration.md:63]: Thanks. I have not understood this.
I assume that in this case, they should also build the test, and not only the main project. Or does "rebuild" rebuild all tests too?

File: docs/kotlin-migration.md (hunk 2 above)
Comment (Arthur-Milchior) [reply docs/kotlin-migration.md:90]: I preferred to refer to the first section from the second one

File: docs/kotlin-migration.md (hunk 1 above)
Comment (david-allison-1) [reply docs/kotlin-migration.md:68]: They need to build the tests

File: docs/kotlin-migration.md (hunk 2 above)
Comment (david-allison-1) [reply docs/kotlin-migration.md]: OK! Not going to bikeshed.

**Review by david-allison-1 — COMMENTED** [review]
Thanks for this, one comment still open, but this looks so much better, thanks for your time

File: docs/kotlin-migration.md (hunk 1 above)
Comment (Arthur-Milchior) [reply docs/kotlin-migration.md]: That's now explicitly specified

**Review by david-allison-1 — APPROVED** [review]
LGTM, merge after build on master passes

**Pull request merged**